CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+# Contributing to jade
+
+### Step 1: Fork
+
+Fork the project [on GitHub](https://github.com/jadejs/jade) and check out your
+copy locally.
+
+```text
+$ git clone git@github.com:username/jade.git
+$ cd jade
+$ git remote add upstream git://github.com/jadejs/jade.git
+```
+
+#### Which branch?
+
+For developing new features and bug fixes, the `master` branch should be pulled
+and built upon.
+
+### Step 2: Branch
+
+Create a feature branch and start hacking:
+
+```text
+$ git checkout -b my-feature-branch -t origin/master
+```
+
+### Step 3: Commit
+
+Make sure git knows your name and email address:
+
+```text
+$ git config --global user.name "J. Random User"
+$ git config --global user.email "j.random.user@example.com"
+```
+
+### Step 4: Rebase
+
+Use `git rebase` (not `git merge`) to sync your work from time to time.
+
+```text
+$ git fetch upstream
+$ git rebase upstream/master
+```
+
+### Step 5: Test
+
+Bug fixes and features **should come with tests**.  Add your tests in the
+test/ directory.  Look at other tests to see how they should be
+structured (license boilerplate, common includes, etc.).
+
+```text
+$ npm test
+```
+
+Make sure that all tests pass.  Please, do not submit patches that fail.
+
+
+### Step 6: Push
+
+```text
+$ git push origin my-feature-branch
+```
+
+Go to https://github.com/yourusername/jades and select your feature branch.
+Click the 'Pull Request' button and fill out the form.
+
+Pull requests are usually reviewed within a few days.  If there are comments
+to address, apply your changes in a separate commit and push that to your
+feature branch.  Post a comment in the pull request afterwards; GitHub does
+not send out notifications when you add commits.
+
+
+## Developer's Certificate of Origin 1.0
+
+By making a contribution to this project, I certify that:
+
+* (a) The contribution was created in whole or in part by me and I
+  have the right to submit it under the open source license indicated
+  in the file; or
+* (b) The contribution is based upon previous work that, to the best
+  of my knowledge, is covered under an appropriate open source license
+  and I have the right under that license to submit that work with
+  modifications, whether created in whole or in part by me, under the
+  same open source license (unless I am permitted to submit under a
+  different license), as indicated in the file; or
+* (c) The contribution was provided directly to me by some other
+  person who certified (a), (b) or (c) and I have not modified it.
+
+
+## Code of Conduct
```

Comment:
I have no personal opinion on this and would defer to those in marginalised groups and those who have spent more time thinking about the issues.  I do think it's important to consider what actions we are willing to take and in what circumstances, especially with regards to harassment that happens offline/in other unrelated spheres.